dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java

```diff
@@ -199,6 +212,74 @@ public long launchPipeline(String entryVal, boolean localEntry, String yamlFile,
+     * @return void errors out if checksums do not match, provides a warning if a remote checksum is null
+     */
+    public void validateDescriptorChecksum(ToolDescriptor.TypeEnum type, String entryVal) {
+        final String[] parts = entryVal.split(":");
+        final String path = parts[0];
+
+        // workflows/checkers have a special prefix for TRS endpoints
+        final String ga4ghv20Path = abstractEntryClient.getTrsId(path);
+
+        // Get the entry version we are validating for
+        final String versionID = abstractEntryClient.getVersionID(entryVal);
+
+        final List<ToolFile> allDescriptors = abstractEntryClient.getAllToolDescriptors(type.toString(), ga4ghv20Path, versionID);
+
+        final Ga4Ghv20Api ga4ghv20api = abstractEntryClient.getClient().getGa4Ghv20Api();
+
+        // All secondary files are located relative to the location of the primary descriptor.
+        final String localTemporaryDirectory = localPrimaryDescriptorFile.getParent();
+        final String checksumFunction = "sha1";
+
+        // Validate each tool file associated with the entry (Primary and secondary descriptors)
+        for (ToolFile toolFile : allDescriptors) {
+
+            // Get remote descriptor checksum
+            Optional<Checksum> remoteDescriptorChecksum = Optional.empty();
+            try {
+                // The TRS endpoint only discovers published entries
+                final FileWrapper remoteDescriptor = ga4ghv20api.toolsIdVersionsVersionIdTypeDescriptorRelativePathGet(type.toString(), ga4ghv20Path, versionID, toolFile.getPath());
+                remoteDescriptorChecksum = remoteDescriptor.getChecksum()
+                    .stream()
+                    .filter(c -> c.getType().equals(checksumFunction))
+                    .findFirst();
+            } catch (io.dockstore.openapi.client.ApiException ex) {
+                exceptionMessage(ex, "unable to locate remote descriptor " + ga4ghv20Path, ENTRY_NOT_FOUND);
+            }
+
+            if (!remoteDescriptorChecksum.isEmpty()) {
+
+                // Get local descriptor checksum
+                Checksum localDescriptorChecksum = null;
+                try {
+                    // if the toolFile.getPath() is absolute, it is converted to a relative path by File the constructor
+                    final File localDescriptor = new File(localTemporaryDirectory, toolFile.getPath());
+                    final String fileContents = FileUtils.readFileToString(localDescriptor, StandardCharsets.UTF_8);
+                    final String fileChecksum = DigestUtils.sha1Hex(fileContents);
+                    localDescriptorChecksum = (new Checksum()).checksum(fileChecksum);
+                    localDescriptorChecksum.setType(checksumFunction);
+                } catch (IOException ex) {
+                    exceptionMessage(ex, "unable to locate local descriptor at " + localTemporaryDirectory + "/" + toolFile.getPath(), IO_ERROR);
```

Comment:
You already have `localDescriptor`, you're sort of reconstructing it here again.

Also I would make this and other error messages sentences, or at least start the phrase with a capital. You could argue it either way, but we should be consistent, and a quick grep shows most of our error messages start with capitals.

Reply:
[Fixed](https://github.com/dockstore/dockstore-cli/blob/2a2f2ea14f509729cf60b324bfd05beb55fac418/dockstore-client/src/main/java/io/dockstore/client/cli/nested/BaseLanguageClient.java#L261).